src/hiero_sdk_python/transaction/transaction.py

```
@@ -262,11 +261,19 @@ def freeze(self):
         if self.transaction_id is None:
             raise ValueError("Transaction ID must be set before freezing. Use freeze_with(client) or set_transaction_id().")
         
-        if self.node_account_id is None:
-            raise ValueError("Node account ID must be set before freezing. Use freeze_with(client) or manually set node_account_id.")
+        if self.node_account_id is None and len(self.node_account_ids) == 0:
+            raise ValueError("Node account ID must be set before freezing. Use freeze_with(client) or manually set node_account_ids.")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)